docs/user/device-audit-logs.md

```diff
@@ -0,0 +1,681 @@
+```
+
+**Example output:**
+```
+2024-11-19T10:00:00Z	dev-01	bootstrap	current	->0
+2024-11-19T10:15:23Z	dev-01	sync	current	0->1
+2024-11-19T11:30:45Z	dev-01	upgrade	desired	1->2
+```
+
+### Integration with Log Aggregation
+
+Audit logs can be shipped to centralized logging systems using standard log forwarders:
+
+#### Example: Fluentd Configuration
+
+```conf
+<source>
+  @type tail
+  path /var/log/flightctl/audit.log
+  pos_file /var/log/flightctl/audit.log.pos
+  tag flightctl.audit
+  <parse>
+    @type json
+  </parse>
+</source>
+
+<match flightctl.audit>
+  @type forward
+  <server>
+    host log-aggregator.example.com
+    port 24224
+  </server>
+</match>
+```
+
+#### Example: Promtail Configuration (Loki)
+
+```yaml
+scrape_configs:
+  - job_name: flightctl-audit
+    static_configs:
+      - targets:
+          - localhost
+        labels:
+          job: flightctl-audit
+          __path__: /var/log/flightctl/audit.log
+```
+
+## Configuration
+
+Audit logging is **enabled by default** and requires no configuration. To explicitly control it, modify the agent configuration:
+
+### Enabling Audit Logging
+
+In `/etc/flightctl/config.yaml`:
+
+```yaml
+audit:
+  enabled: true
+```
+
+### Disabling Audit Logging
+
+```yaml
+audit:
+  enabled: false
+```
+
+> [!WARNING]
+> Disabling audit logs removes visibility into device state changes and may impact compliance requirements.
+
+## Log Location and Permissions
+
+| Property     | Value                              |
+|--------------|------------------------------------|
+| **Path**     | `/var/log/flightctl/audit.log` (hardcoded, not configurable) |
+| **Format**   | JSONL (one JSON object per line)   |
+
+The directory `/var/log/flightctl` is created automatically by the agent when writing the first audit event.
+
+## Automatic Log Rotation
+
+Audit logs use **automatic size-based rotation** built into the agent (using lumberjack). No external log rotation configuration is needed.
+
+### Rotation Settings
+
+These settings are hardcoded and not user-configurable:
+
+| Setting         | Value                                      |
+|-----------------|--------------------------------------------|
+| **Max Size**    | 1 MB per file |
+| **Max Backups** | 3 backup files                             |
+| **Max Age**     | No time-based pruning                      |
+| **Compression** | Enabled (rotated files are compressed)     |
+| **Total Space** | ≈4 MB (1 active + 3 backups)               |
+| **Capacity**    | ~3,495 records per file, ~13,980 records total |
+
+### Rotation Behavior
+
+- When `/var/log/flightctl/audit.log` reaches 1 MB, it's automatically rotated
+- Current file renamed to `audit.log.1` (compressed to `audit.log.1.gz`)
+- Previous backups shift: `audit.log.1` → `audit.log.2` → `audit.log.3`
+- Oldest backup (`audit.log.3`) is deleted
+- New empty `audit.log` file created
+- No agent restart or reload required
```

Comment:
can we reduce this to one concise sentence that we automatically handle rotation and compression